src/lib.rs

```diff
@@ -433,7 +433,8 @@ impl Connection<Authenticated> {
             let mut w = wbuf.lock().unwrap();
             w.in_flush = false;
             if cur_len > 0 {
-                if let Err(_) = w.writer.flush() {
+                if let Err(error) = w.writer.flush() {
+                    eprintln!("Flusher thread failed to flush: {:?}", error);
```

Comment:
Again maybe capture for possible last_error(), eventually never println from production client lib.

Reply:
What do you have in mind when you say "capture"? One thing I like to do with code that runs in one thread but causes functionality associated with methods called elsewhere to misbehave is to set an AtomicPtr to refer to an error, and have all caller threads check the error pointer to see if it's null or if it points to something, which will usually be really cheap, but will feed the error asynchronously back to the callers. Are you thinking something similar?

Reply:
For async errors from server was thinking last_error() on Connection and for callbacks for subs with handler same name but just for the handler based subscriptions.